torchrl/objectives/value/utils.py

```diff
@@ -215,8 +217,10 @@ def _get_num_per_traj(dones_and_truncated):
     return num_per_traj
 
 
-def _split_and_pad_sequence(tensor, splits):
-    """Given a tensor of size [B, T, *other] and the corresponding traj lengths (flattened), returns the padded trajectories [NPad, Tmax, *other].
+def _split_and_pad_sequence(
+    tensor: Union[torch.Tensor, TensorDictBase], splits: torch.Tensor
+):
+    """Given a tensor of size [*B, T, F] and the corresponding traj lengths (flattened), returns the padded trajectories [NPad, Tmax, *other].
```

Comment:
if *other equal to F?

Reply:
Yes. The new version behaves as the previous one on all input shapes, or at least on all input shapes I've tested.
I just wanted to streamline the docu with other functions in functional.py.

Moreover, in practice all methods using `_split_and_pad_sequence` must also use `_get_num_per_traj` which needs input in the form of [*B, T]. Thus in all methods the input is anyway reshaped (`vec_generalized_advantage_estimate`, `reward2go`).